yash-datalake/nested/dataFactory.json

```diff
@@ -1,293 +1,306 @@
-				"dependsOn": [
-					"[parameters('dataFactoryName')]",
-					"[variables('clientAzureStorageLinkedServiceName')]",
-					"[variables('onDemandHDInsightLinkedServiceName')]"
-				],
-				"apiVersion": "2017-09-01-preview",
-				"properties": {
-					"activities": [{
-						"name": "MySparkActivity",
-						"type": "HDInsightSpark",
-						"dependsOn": [],
-						"policy": {
-							"timeout": "7.00:00:00",
-							"retry": 0,
-							"retryIntervalInSeconds": 30
-						},
-						"typeProperties": {
-							"rootPath": "[concat(variables('blobContainer'), '/jars')]",
-							"entryFilePath": "[variables('entryFilePath')]",
-							"arguments": [
-								"[parameters('storageAccountName')]",
-								"[parameters('KibanaESDeployedVMIP')]"
-							],
-                            "className": "com.neo4j.AzureNeo4J",
-							"sparkJobLinkedService": {
-								"referenceName": "[variables('clientAzureStorageLinkedServiceName')]",
-								"type": "LinkedServiceReference"
-							}
-						},
-						"linkedServiceName": {
-							"referenceName": "[variables('onDemandHDInsightLinkedServiceName')]",
-							"type": "LinkedServiceReference"
-						}
-					}]
-				}
-			}
-		]
-	}]
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "KibanaESDeployedVMIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the data factory. Must be globally unique."
+      }
+    },
+    "dataFactoryName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the data factory. Must be globally unique."
+      }
+    },
+    "dataFactoryLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "Location of the data factory. Currently, only East US and East US 2 are supported. "
+      }
+    },
+    "blobContainer": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the blob container in the Azure Storage account."
+      }
+    },
+    "clientAzureStorageConnectionString": {
+      "type": "string",
+      "metadata": {
+        "description": "Connection string for the Azure Storage account."
+      }
+    },
+    "servicePrincipalId": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "servicePrincipalKey": {
+      "type": "string",
+      "metadata": {
+        "description": "The access key of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "storage account name"
+      }
+    }
+  },
+  "variables": {
+    "_artifactsLocation": "https://yashinnovationstorage.blob.core.windows.net/yash-datalake-resource",

```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts